plugins/chat/plugin.rb

```diff
@@ -150,6 +150,14 @@ module ::Chat
     scope.can_direct_message? && Guardian.new(object).can_chat?
   end
 
+  add_to_serializer(:hidden_profile, :can_chat_user) do
+    return false if !SiteSetting.chat_enabled
+    return false if scope.user.blank? || scope.user.id == object.id
```

Comment:
No, I don't think so. While you can message yourself, showing a button in your profile to message yourself seems more like a bug to me than something I would expect. `scope.user.id == object.id` is redundant anyway since we don't use the HiddenProfile serializer for our own profile